web/src/main/java/no/nav/modiapersonoversikt/config/MetricsConfig.kt

```diff
@@ -0,0 +1,49 @@
+package no.nav.modiapersonoversikt.config
+
+import io.micrometer.core.instrument.binder.okhttp3.OkHttpMetricsEventListener
+import io.micrometer.prometheus.PrometheusMeterRegistry
+import no.nav.common.rest.client.RestClient
+import no.nav.modiapersonoversikt.service.unleash.Feature
+import no.nav.modiapersonoversikt.service.unleash.UnleashService
+import no.nav.modiapersonoversikt.utils.MaskingUtils
+import no.nav.modiapersonoversikt.utils.UnleashProxySwitcher
+import okhttp3.Request
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.annotation.Configuration
+import javax.annotation.PostConstruct
+
+@Configuration
+open class MetricsConfig {
+    @Autowired
+    lateinit var prometheusMeterRegistry: PrometheusMeterRegistry
+
+    @Autowired
+    lateinit var unleash: UnleashService
+
+    @PostConstruct
+    fun checkIfMetricsListenerShouldBeEnabled() {
+        return UnleashProxySwitcher.createSwitcher(
+            featureToggle = Feature.USE_REST_CLIENT_METRICS,
+            unleashService = unleash,
+            ifEnabled = initRestClientWithMetricsListener(),
```

Comment:
Burde nok heller ha vært noe alà;
```kotlin
@PostConstruct
fun setup() {
  if (unleash.isEnabled(Feature.USE_REST_CLIENT_METRICS)) {
    initRestClientWithMetricsListener()
  }
}
```


`UnleashProxySwitcher` er tiltenkt å brukes for å bytte mellom to underliggende tjenester.

For eksempel:
```kotlin
interface MyAwesomeService {
  fun getAwesomeQuote(): String
  fun getAwesomeNumber(): String
}
val restService: MyAwesomeService = TODO()
val soapService: MyAwesomeService = TODO()

val service = UnleashProxySwitcher.createSwitcher(
  featureToggle = Feature.USE_REST_SERVICE,
  unleashService = unleashService,
  ifEnabled = restService,
  ifDisabled = soapService
)

println(service.getAwesomeQuote())
println(service.getAwesomeNumber())
```

Reply:
Ah, takk, det gir mer mening! Synes switchen virket litt vel komplisert for den funksjonaliteten jeg trengte 😛